src/event/log_event.rs

```diff
@@ -3,80 +3,91 @@ use serde::{Serialize, Serializer};
 use std::{
     collections::{btree_map::Entry, BTreeMap, HashMap},
     convert::{TryFrom, TryInto},
+    fmt::{Debug, Display},
     iter::FromIterator,
 };
-use string_cache::DefaultAtom;
 
 #[derive(PartialEq, Debug, Clone, Default)]
 pub struct LogEvent {
     fields: BTreeMap<String, Value>,
 }
 
 impl LogEvent {
-    pub fn get(&self, key: &DefaultAtom) -> Option<&Value> {
-        util::log::get(&self.fields, key)
+    #[instrument(skip(self, key), fields(key = %key.as_ref()))]
```

Comment:
I see you've been adding these since the introduction of the `Lookup` type. Do you recommend we add them to all methods? Or just certain ones? They seem useful for tracing.

Reply:
We haven't really made an architectural call one way or the other. What they do is populate the span: https://tracing-rs.netlify.app/tracing/attr.instrument.html, so if we did wide adoption we might be able to reap some really nice contextual rewards. It's a practice I've been doing in my own code lately. I'm not sure if @lukesteensen has an opinion though.

Reply:
I've seen these used as well.

I agree that better instrumentation is a worthy goal, but I would prefer if we at least document how/why we use them, including some best-practices/patterns everyone should follow.

I don't mind if it's in an RFC or a "tracking issue" that contains some context, and a few relevant links.

Reply:
> better instrumentation is a worthy goal, but I would prefer if we at least document how/why we use them, including some best-practices/patterns everyone should follow.

I agree! I'd also note that we should strive to be as consistent as possible across the codebase. The macro is cool and useful, but right now it mostly tells me which code was written by @Hoverbear :smile: 

An easy first step would be to open an issue for setting some guidelines around thing kind of instrumentation. We'll want to cover `#[instrumented(...)]` vs `trace!` vs `emit!`, message style, etc in the eventual guidelines, but for now, maybe @Hoverbear can just summarize her current thoughts as a starting point for the conversation.

Reply:
I'd love to! For now, as a note `#[instrument]` just creates a span when the function is entered, with the function args (or as configured) as span items. It's dropped when the function exits. That's all it does. :)

It doesn't log. :)